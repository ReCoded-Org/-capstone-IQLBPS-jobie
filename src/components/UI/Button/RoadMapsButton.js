const RoadMapsButton = ({ svg, title, link }) => {
  return (
    <a
      href={link}
      target="_blank"
      rel="noreferrer"
      className="small:w-20 small:h-20 w-36 h-32 medium:w-32 medium:h-28 pt-9 medium:pt-7 small:pt-4 shadow-md bg-white rounded-lg hover:scale-105 hover:shadow-lg"
    >
      <svg
        xmlns={svg.xmlns}
        viewBox={svg.viewBox}
        fill="#EC7063"
        className="w-7 medium:w-6 small:w-4 mx-auto mb-1"
      >
        <path d={svg.d} />
      </svg>

      <p className="text-center text-lg medium:text-base small:text-sm">
        {title}
      </p>
    </a>
  );
};
export default RoadMapsButton;
