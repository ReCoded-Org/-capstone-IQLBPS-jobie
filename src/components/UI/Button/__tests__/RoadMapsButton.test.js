import React from "react";
import renderer from "react-test-renderer";
import RoadMapsButton from "../RoadMapsButton";
const data = {
  svg: {
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 384 512",
    d: "M0 32l34.9 395.8L191.5 480l157.6-52.2L384 32H0zm308.2 127.9H124.4l4.1 49.4h175.6l-13.6 148.4-97.9 27v.3h-1.1l-98.7-27.3-6-75.8h47.7L138 320l53.5 14.5 53.7-14.5 6-62.2H84.3L71.5 112.2h241.1l-4.4 47.7z",
  },
  link: "https://roadmap.sh/frontend",
  title: "Frontend",
  id: 1,
};

it("renders Road Map button correctly", () => {
  const tree = renderer
    .create(
      <RoadMapsButton svg={data.svg} link={data.link} title={data.title} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
