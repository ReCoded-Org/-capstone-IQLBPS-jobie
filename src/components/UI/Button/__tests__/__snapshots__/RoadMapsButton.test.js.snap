// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Road Map button correctly 1`] = `
<a
  className="small:w-20 small:h-20 w-36 h-32 medium:w-32 medium:h-28 pt-9 medium:pt-7 small:pt-4 shadow-md bg-white rounded-lg hover:scale-105 hover:shadow-lg"
  href="https://roadmap.sh/frontend"
  rel="noreferrer"
  target="_blank"
>
  <svg
    className="w-7 medium:w-6 small:w-4 mx-auto mb-1"
    fill="#EC7063"
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 32l34.9 395.8L191.5 480l157.6-52.2L384 32H0zm308.2 127.9H124.4l4.1 49.4h175.6l-13.6 148.4-97.9 27v.3h-1.1l-98.7-27.3-6-75.8h47.7L138 320l53.5 14.5 53.7-14.5 6-62.2H84.3L71.5 112.2h241.1l-4.4 47.7z"
    />
  </svg>
  <p
    className="text-center text-lg medium:text-base small:text-sm"
  >
    Frontend
  </p>
</a>
`;
